### PR TITLE
Add rate my professor data integration to course reviews

### PR DESCRIPTION
**Pull request by Richie78321**
Add rate my professor data to the course testimonial section using a helper Python script and a [fork of the Pitt API](https://github.com/Richie78321/PittAPI) that contains an RMP scraper.

### CODE REVIEW

File: src/components/professor-link.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  return (
+    <a 
+      href={`https://www.ratemyprofessors.com/ShowRatings.jsp?tid=${profRMPData.professor_id}`}
+      target="_blank"
+      rel="noopener noreferrer"
+      className="tooltip text-black">
+      {profRMPData.professor_full_name}
+      <div className="bg-gray-100 tooltiptext">
```
Comment (varughese) [inline src/components/professor-link.js:37]: update to 

`bg-white block border px-4 py-3 shadow-2xl tooltip`

image

delete `<p className="mb-0 text-sm font-normal text-gray-500">Click to see more</p>` for now

File: src/styles/components.scss [hunk 2]
```diff
@@ -275,4 +275,38 @@ header {
...
+
+.tooltip .tooltiptext::after {
+  content: "";
+  position: absolute;
+  top: 100%;
+  left: 50%;
+  margin-left: -5px;
+  border-width: 5px;
+  border-style: solid;
+  border-color: #F7FAFC transparent transparent transparent;
```
Comment (varughese) [inline src/styles/components.scss:307]: this triangle with css thing is one of the best tricks in the book

update to

```
content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    margin-left: -15px;
    border: 15px solid transparent;
    border-top-color: #ffffff;
}
```

File: src/styles/components.scss [hunk 3]
```diff
@@ -275,4 +275,38 @@ header {
...
+}
+
+.tooltip {
+  position: relative;
+  display: inline-block;
+  border-bottom: 1px dotted black;
+}
+
+.tooltip .tooltiptext {
+  visibility: hidden;
```
Comment (varughese) [inline src/styles/components.scss:287]: change to:

visibility: hidden;
    width: 200px;
    z-index: 1;
    bottom: 150%;
    left: 50%;
    margin-left: -60px;

**Review by varughese — CHANGES_REQUESTED** [review]

File: helpers/rmp/get-rmp-data.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Ideally in the future the Pitt API will be hosted so it doesn't need to exist locally.
+# Currently requires fork of Pitt API with RMP scraper: https://github.com/Richie78321/PittAPI
+from pittapi import ratemyprofessors
```
Comment (varughese) [inline helpers/rmp/get-rmp-data.py:3]: can we add a script to package.json that sets up and runs this? 

so someone could do like

```
npm run generate_rmp_data_json
```

File: src/components/testimonial.js [hunk 5]
```diff
@@ -13,7 +14,7 @@ const Testimonial = ({ review, difficulty, prof, term, year, quality }) => {
           </span>
           <span>
             <span className="text-gray-500">Professor </span>
-            <span className="font-bold"> {prof} </span>
+            <span className="font-bold"> <ProfessorLink prof={prof} /> </span>
```
Comment (varughese) [inline src/components/testimonial.js:17]: lovin the API for this

File: src/components/professor-link.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { useStaticQuery, graphql } from "gatsby";
+import React from "react"
+
+const ProfessorLink = ({ prof }) => {
```
Comment (varughese) [inline src/components/professor-link.js:4]: nit (can choose to ignore): id probably call this ProfessorTooltip

**Review by varughese — APPROVED** [review]

**Comment by varughese** [comment]
great feature and solid code, from the api to the frontend change. Good stuff

File: src/components/professor-link.js (hunk 6 above)
Comment (Richie78321) [reply src/components/professor-link.js:4]: Agreed. Done

File: helpers/rmp/get-rmp-data.py (hunk 4 above)
Comment (Richie78321) [reply helpers/rmp/get-rmp-data.py:3]: Yes good idea I'll look into this

Comment (Richie78321) [reply helpers/rmp/get-rmp-data.py]: Done!

**Pull request merged**